runtime/src/accounts_db.rs

```diff
@@ -4364,6 +4365,23 @@ impl AccountsDb {
                             });
                         }
                     }
+                    if let Some(cache) = accounts_cache {
+                        if let Some(slot_cache) = cache.slot_cache(slot) {
+                            let keys = slot_cache.get_all_pubkeys();
```

Comment:
Would it be beneficial/faster to have `get_all_pubkeys()` return `Vec<&Pubkey>` instead? It looks like sometimes the loop below might not need the copy. (And then it'd be simple to just `*key` in the `let mut accessor = ...` call.)

Reply:
get_all_pubkeys reads keys out of a dashmap. The key values are behind a lock structure. I do not imagine it is straightforward or even possible to get a reference without holding a lock open. There are other ways to do this as well, adding an iterate function perhaps to the SlotCache. This one seemed safe and straightforward, but I'm open to alternative suggestions.

Reply:
It's not possible. To hold a reference to the keys you would have to hold the lock open.

Reply:
Gotcha. Thanks for the info!